worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,373 @@
+    options: YoshisIslandOptions
+
+    locked_locations: List[str]
+    location_cache: List[Location]
+    set_req_bosses: str
+    lives_high: int
+    lives_low: int
+    castle_bosses: int
+    bowser_bosses: int
+    baby_mario_sfx: int
+    leader_color: int
+    boss_order: list
+    boss_burt: int
+    luigi_pieces: int
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        self.rom_name_available_event = threading.Event()
+        super().__init__(multiworld, player)
+        self.locked_locations= []
+        self.location_cache= []
+
+    @classmethod
+    def stage_assert_generate(cls, multiworld: MultiWorld) -> None:
+        rom_file = get_base_rom_path()
+        if not os.path.exists(rom_file):
+            raise FileNotFoundError(rom_file)
+
+    def fill_slot_data(self) -> Dict[str, List[int]]:
+        return {
+            "world_1": self.world_1_stages,
+            "world_2": self.world_2_stages,
+            "world_3": self.world_3_stages,
+            "world_4": self.world_4_stages,
+            "world_5": self.world_5_stages,
+            "world_6": self.world_6_stages
+        }
+
+    def write_spoiler_header(self, spoiler_handle: TextIO) -> None:
+        spoiler_handle.write(f"Burt The Bashful's Boss Door:      {self.boss_order[0]}\n")
+        spoiler_handle.write(f"Salvo The Slime's Boss Door:       {self.boss_order[1]}\n")
+        spoiler_handle.write(f"Bigger Boo's Boss Door:            {self.boss_order[2]}\n")
+        spoiler_handle.write(f"Roger The Ghost's Boss Door:       {self.boss_order[3]}\n")
+        spoiler_handle.write(f"Prince Froggy's Boss Door:         {self.boss_order[4]}\n")
+        spoiler_handle.write(f"Naval Piranha's Boss Door:         {self.boss_order[5]}\n")
+        spoiler_handle.write(f"Marching Milde's Boss Door:        {self.boss_order[6]}\n")
+        spoiler_handle.write(f"Hookbill The Koopa's Boss Door:    {self.boss_order[7]}\n")
+        spoiler_handle.write(f"Sluggy The Unshaven's Boss Door:   {self.boss_order[8]}\n")
+        spoiler_handle.write(f"Raphael The Raven's Boss Door:     {self.boss_order[9]}\n")
+        spoiler_handle.write(f"Tap-Tap The Red Nose's Boss Door:  {self.boss_order[10]}\n")
+        spoiler_handle.write(f"\nLevels:\n1-1: {self.level_name_list[0]}\n")
+        spoiler_handle.write(f"1-2: {self.level_name_list[1]}\n")
+        spoiler_handle.write(f"1-3: {self.level_name_list[2]}\n")
+        spoiler_handle.write(f"1-4: {self.level_name_list[3]}\n")
+        spoiler_handle.write(f"1-5: {self.level_name_list[4]}\n")
+        spoiler_handle.write(f"1-6: {self.level_name_list[5]}\n")
+        spoiler_handle.write(f"1-7: {self.level_name_list[6]}\n")
+        spoiler_handle.write(f"1-8: {self.level_name_list[7]}\n")
+
+        spoiler_handle.write(f"\n2-1: {self.level_name_list[8]}\n")
+        spoiler_handle.write(f"2-2: {self.level_name_list[9]}\n")
+        spoiler_handle.write(f"2-3: {self.level_name_list[10]}\n")
```

Comment:
Are you sure you don't want to add `"King Bowser's Castle"` to the `level_name_list`
and then do something like this?

```python
        spoiler_handle.write("\nLevels:")
        for level_index in range(48):
            world = level_index // 8 + 1
            level = level_index % 8 + 1
            if level == 1:
                spoiler_handle.write("\n")
            spoiler_handle.write(f"{world}-{level}: {self.level_name_list[level_index]}\n")
```
...or not add `"King Bowser's Castle"` to the list
and `range(47)`